lib/ts/recipe/accountlinking/recipe.ts

```diff
@@ -17,82 +17,553 @@ import error from "../../error";
+        }
+        if (info.phoneNumber !== undefined) {
+            return identitiesForPrimaryUser.verified.phoneNumbers.includes(info.phoneNumber);
+        }
+        throw Error("it should never reach here");
     };
-    createPrimaryUserIdOrLinkAccountPostSignUp = async (_input: {
-        identifyinInfo: AccountInfoAndEmailWithRecipeId;
-        shouldRequireVerification: boolean;
-    }) => {
-        // TODO
+    createPrimaryUserIdOrLinkAccountPostSignUp = async ({
+        info,
+        infoVerified,
+        recipeUserId,
+        userContext,
+    }: {
+        info: AccountInfoAndEmailWithRecipeId;
+        infoVerified: boolean;
+        recipeUserId: string;
+        userContext: any;
+    }): Promise<string> => {
+        let shouldDoAccountLinking = await this.config.shouldDoAutomaticAccountLinking(
+            info,
+            undefined,
+            undefined,
+            userContext
+        );
+        if (shouldDoAccountLinking.shouldAutomaticallyLink && shouldDoAccountLinking.shouldRequireVerification) {
+            if (!infoVerified) {
+                return recipeUserId;
+            }
+        }
+        let canCreatePrimaryUserId = await this.recipeInterfaceImpl.canCreatePrimaryUserId({
+            recipeUserId,
+            userContext,
+        });
+        if (canCreatePrimaryUserId) {
+            let user = await this.recipeInterfaceImpl.createPrimaryUser({
+                recipeUserId,
+                userContext,
+            });
+            if (user.status !== "OK") {
+                throw Error(user.status);
+            }
+            return user.user.id;
+        }
+        let identifier:
+            | {
+                  email: string;
+              }
+            | {
+                  phoneNumber: string;
+              };
+        if (info.email !== undefined) {
+            identifier = {
+                email: info.email,
+            };
+        } else if (info.phoneNumber !== undefined) {
+            identifier = {
+                phoneNumber: info.phoneNumber,
+            };
+        } else {
+            throw Error("this error should never be thrown");
+        }
+        let users = await this.recipeInterfaceImpl.listUsersByAccountInfo({
+            info: identifier,
+            userContext,
+        });
+        if (users === undefined || users.length === 0) {
+            throw Error("this error should never be thrown");
+        }
+        let primaryUser = users.find((u) => u.isPrimaryUser);
+        if (primaryUser === undefined) {
+            throw Error("this error should never be thrown");
+        }
+        shouldDoAccountLinking = await this.config.shouldDoAutomaticAccountLinking(
+            info,
+            primaryUser,
+            undefined,
+            userContext
+        );
+        if (!shouldDoAccountLinking.shouldAutomaticallyLink) {
+            return recipeUserId;
+        }
+        await this.recipeInterfaceImpl.linkAccounts({
```

Comment:
do not ignore the status returned from this function. You can throw a should never come here error in case this does not return OK